### PR TITLE
Widget update fixes

### PR DESCRIPTION
**Pull request by tych0**
Here's some more widget update fixes, which should ultimately result in fewer bar draws. As a bonus, the code is simplified, and all _TextBox based widgets get truncation functionality.

### CODE REVIEW

File: libqtile/widget/currentscreen.py [hunk 1]
```diff
@@ -45,15 +45,11 @@ def _configure(self, qtile, bar):
...
-            self.text = self.inactive_text
-            self.foreground = self.inactive_color
+            if self.qtile.current_screen == self.bar.screen:
+                self.foreground = self.active_color
+                self.update(self.active_text)
+            else:
+                self.foreground = self.inactive_color
+                self.update(self.inactive_text)
+
+        hook.subscribe.current_screen_change(self.hook_response)
```
Comment (m-col) [inline libqtile/widget/currentscreen.py:55]: Is there any reason `hook_response` is dynamically constructed within `setup_hooks` and not a class method? Looks like you hooked `self.hook_response` as if it was.

Comment (tych0) [reply libqtile/widget/currentscreen.py:55]: Oh, I think I had it that way before and didn't fix this. I can make it a class method, thanks for catching this.

File: test/widgets/test_window_count.py [hunk 2]
```diff
@@ -1,70 +1,38 @@
...
-    fakebar = Bar([wc], 24)
-    wc.bar = fakebar
-    qtile = FakeQtile()
-    wc.qtile = qtile
-    wc._setup_hooks()
+    manager_nospawn.start(WindowCountConf)
 
     # No windows opened
-    assert wc._count == 0
+    assert int(manager_nospawn.c.widget["windowcount"].get()) == 0
```
Comment (m-col) [inline test/widgets/test_window_count.py:17]: do these need the `int`s?

**Review by m-col — APPROVED** [review]
Nice, looks good, especially this: 
image

File: test/widgets/test_window_count.py (hunk 2 above)
Comment (tych0) [reply test/widgets/test_window_count.py:17]: I think so; the return value is a string.

Comment (m-col) [reply test/widgets/test_window_count.py:17]: Ah yep, of course.

**Comment by m-col** [comment]
Cool cool, LGTM!

**Comment by ramnes** [comment]
Excellent, thank you!

**Pull request merged**